### DISCUSSION TITLE
Configuring LTHEORY build

### DISCUSSION THREAD

**Discussion by FormalMoss** [JoshParnell/ltheory#5] [General]
As per instructions on https://github.com/JoshParnell/ltheory
AND being a non-programmer, I installed the MSI entitled "cmake-3.24.0-windows-x86_64"

The install directory for my LTheory folder is
 ~/Desktop/_LT1/ltheory

and when I attempt to run step 5, i.e. "python configure.py", i get the following errors, which prevent me from continuing

$ python configure.py
Traceback (most recent call last):
  File "configure.py", line 21, in <module>
    subprocess.call(['cmake', '-S', './', '-B', './build'])
  File "C:\_Python38-32\lib\subprocess.py", line 340, in call
    with Popen(*popenargs, **kwargs) as p:
  File "C:\_Python38-32\lib\subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\_Python38-32\lib\subprocess.py", line 1307, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

Can anyone help me on where I went wrong/ which version of cmake I "should" have installed??
Kudos to Cornflakes for figuring this out, as he is usually the one to highlight my obvious issues known to all :) :P 

P.S. I followed Josh's instructions to the letter, unfortunately there were 3x options to choose for cmake:
- cmake-3.24.0-windows-x86_64.zip
- cmake-3.24.0-windows-x86_64.msi
- cmake-3.24.0.zip
I am thinking that I may require the 32-bit version of cmake, but err'ing on the side of caution here, as some (hopefully I am the only one) may encounter similar issues in future..?

**Comment by JoshParnell** [comment]
Hey Formal, I think it's not an issue of version, but of having `cmake` as part of your system path.

In the installer, there's an option for this somewhere. You need to make sure you select "add to system path", at least for the current user, but might as well do it for all users to be safe. Here's an image I found:

![image](https://user-images.githubusercontent.com/16365885/185165573-59cc41d9-5554-4cb4-9e7c-d9b90921c3c4.png)

Try reinstalling cmake and choosing this option. Sorry that I didn't mention this in the instructions!
